### ISSUE TITLE
Rules in config.json do not work

### ISSUE THREAD

**Issue by neildeadman** [Browsers-software/browsers#88]
**Describe the bug**
Rules in config.json do not work

**System:**
 - OS: macOS 14.0 and Windows 11
 - Browsers Version [e.g. 0.22.1]: 0.4.1

**macOS config:**
`{
  "hidden_apps": [],
  "hidden_profiles": [
    "/Applications/Firefox.app/Contents/MacOS/firefox#default",
    "/System/Volumes/Preboot/Cryptexes/App/System/Applications/Safari.app/Contents/MacOS/Safari#"
  ],
  "profile_order": [],
  "default_profile": null,
  "rules": [
    {
      "url_pattern": "https://www.google.com/",                 
      "profile": "/System/Volumes/Preboot/Cryptexes/App/System/Applications/Safari.app/Contents/MacOS/Safari"
    }
  ],
  "ui": {
    "show_hotkeys": true,
    "quit_on_lost_focus": false
  }
}
`

**Windows 11 config:**
`{
  "hidden_apps": [],
  "hidden_profiles": [
    "C:\\Program Files\\Mozilla Firefox\\firefox.exe#default"
  ],
  "profile_order": [],
  "default_profile": null,
  "rules": [
  {
    "url_pattern": "https://www.google.com/",
    "profile": "C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe"
  }
  ],
  "ui": {
    "show_hotkeys": true,
    "quit_on_lost_focus": false
  }
}`

**Comment by liias** [comment]
Hey there.
It's not obvious, but profile also needs to have # in the end (basically like it is in the profile order list)

**Comment by neildeadman** [comment]
OK, that has sort of helped. Safari now loads when I click my test URL of https://www.google.com/ but I cannot get Edge to work at all. Not tried Chrome yet!

The caches file says my profile name is my 'name', but Edge itself says its just Work. Neither work to open a URL without being asked for which browser to use.

Firefox works with profile named default-release.

**Comment by neildeadman** [comment]
The Edge above was on Mac. On Windows 11 the Ege profile isn't showing a name so ending with just # worked.

**Comment by liias** [comment]
Chrome profile id is actually dir name. You can see all profile names in the cache file (can find path in About menu of Browsers).

I conaiser thia as another vote for providing UI for settings.

**Comment by neildeadman** [comment]
100% a vote for UI! 

I'm not sure what you mean by "dir name" though. I have been using the installed_browsers.json file already.

Here are the entries from that file for Chrome and Edge on my Mac:

`  {
    "command": [
      "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
    ],
    "executable_path": "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
    "display_name": "Chrome",
    "bundle": "com.google.Chrome",
    "user_dir": "/Users/nmd/Library/Application Support/Google/Chrome",
    "icon_path": "/Users/nmd/Library/Caches/software.Browsers/icons/com.google.Chrome.png",
    "profiles": [
      {
        "profile_cli_arg_value": "Default",
        "profile_cli_container_name": null,
        "profile_name": "Neil (Neil McKirdy-Deadman)",
        "profile_icon": "/Users/nmd/Library/Caches/software.Browsers/icons/profiles/com.google.Chrome/Neil (Neil McKirdy-Deadman).png",
        "profile_restricted_url_patterns": []
      }
    ],
    "restricted_domains": []
  },
  {
    "command": [
      "/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge"
    ],
    "executable_path": "/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge",
    "display_name": "Microsoft Edge",
    "bundle": "com.microsoft.edgemac",
    "user_dir": "/Users/nmd/Library/Application Support/Microsoft Edge",
    "icon_path": "/Users/nmd/Library/Caches/software.Browsers/icons/com.microsoft.edgemac.png",
    "profiles": [
      {
        "profile_cli_arg_value": "Default",
        "profile_cli_container_name": null,
        "profile_name": "Neil McKirdy-Deadman",
        "profile_icon": "/Users/nmd/Library/Caches/software.Browsers/icons/profiles/com.microsoft.edgemac/Neil McKirdy-Deadman.png",
        "profile_restricted_url_patterns": []
      }
    ],
    "restricted_domains": []
  },
`

**Comment by liias** [comment]
Sorry for coming back for this so late. If there is a profile, you need to use the value present in `profile_cli_arg_value`.
So the full identifier is `executable_path#profile_cli_arg_value`.

So in your case `/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge#Default`.
Let me know if this fixes it.

In Windows profiles are not well supported yet. Thanks for reminding about this, will add internal task.

**Comment by neildeadman** [comment]
Sorry, somehow missed your update. Will give this a try and report back...

**Comment by neildeadman** [comment]
I seem to have this working well on macOS, but on Windows I cannot seem to get those files create anymore :(

I wiped away the install to start again with the latest version.

This could def use a config app to manage all this.

**Comment by neildeadman** [comment]
I've managed to install and get those files created now by using an older release and then upgrading to the latest release. If I open the app it does now also show the choice of browsers, where as before it did nothing.

However, if I set browsers as the default app for opening links, I get and error window with the URL as the title and in the main body of the error window with th error message **Application not found**. This could be unique to me and not a bug in the app itself, but I am not sure how to work around this. The browsers.log file is empty.

Let me know if this should be a new incident.
